students/bong/users/views.py

```diff
@@ -28,25 +29,25 @@ def post(self, request):
             return JsonResponse({"message" : "DUPLICATION_ERROR"}, status=400)
 
         User.objects.create(
-            name, email, password, phone_num,
-            hashed_password  = (bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt())).decode('utf-8'),
+            name, email, phone_num,
+            password    = (bcrypt.hashpw(data.get("password").encode('utf-8'), bcrypt.gensalt())).decode('utf-8'),
             )
         
-        return JsonResponse({"message" : "SUCCESS"}, status=201)
+        return JsonResponse({"message" : "SUCCESS"}, status=200)
 
 class LoginView(View):
     def post(self, request):
-        data        = json.loads(request.body)
-        email       = data.get("email")
-        password    = data.get("password")
-
+        data            = json.loads(request.body)
+        email           = data.get("email")
+        password        = data.get("password")
+        
         if not (email and password):
             return JsonResponse({"messgae" : "KEY_ERROR"}, status=400)
-
-        if not User.objects.filter(email=email, password=password):
-            return JsonResponse({"message" : "INVALID_USER"}, status=401)
-            
-        return JsonResponse({"message" : "SUCCESS"}, status=200)
-
-        
-
+       
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email=email)
+            if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
+                token = jwt.encode({"email" : email}, SECRET_KEY, algorithm='HS256')
+                return JsonResponse({"TOKEN" : token}, status=200)
+                
+        return JsonResponse({"message" : "INVALID_USER"}, status=401)            
```

Comment:
1. 에러 처리를 할 때에는 if문 보다는 try, except를 사용하는 것이 가독성 측면에서 좋을 수 있습니다.
   if문을 `try + except User.DoesNotExist:` 문으로 바꿀 수 있습니다. 공식문서에서 한번 찾아보세요~
2. 'HS256' 알고리즘은 my_settings.py에 환경변수로 등록해서 관리해주세요! 
3. my_settings.py에서 환경변수를 import해서 사용하실 때에는 지금처럼 `from my_settings import ALGORITHM, SECRET_KEY` 방법이 아니라 `from django.conf.settings import  ALGORITHM, SECRET_KEY`를 사용해서 환경변수를 불러와서 사용해주세요. (django에서 추천하는 방법)

```suggestion
        try:
            user = User.objects.get(email=email)
            
            if bcrypt.checkpw(password.encode('utf-8'), user.password.encode('utf-8')):
                token = jwt.encode({"id" : user.id}, SECRET_KEY, algorithm=ALGORITHM)
                
                return JsonResponse({"TOKEN" : token}, status=200)
                
         except User.DoesNotExist:
            return JsonResponse({"message" : "USER_DOES_NOT_EXISTS"}, status=401)            
```